static/app/views/insights/cache/components/samplePanel.tsx

```diff
@@ -448,3 +440,38 @@ const CACHE_STATUS_OPTIONS = [
     label: t('Miss'),
   },
 ];
+
+export const useTransactionDuration = ({transaction}: {transaction: string}) => {
+  const useEap = useInsightsEap();
+
+  const metricsResult = useMetrics(
+    {
+      enabled: !useEap && Boolean(transaction),
+      search: MutableSearch.fromQueryObject({
+        transaction,
+      } satisfies MetricsQueryFilters),
+      fields: [`avg(${MetricsFields.TRANSACTION_DURATION})`],
+    },
+    Referrer.SAMPLES_CACHE_TRANSACTION_DURATION
+  );
+
+  const eapResult = useEAPSpans(
+    {
+      search: MutableSearch.fromQueryObject({
+        transaction,
+        is_transaction: 'true',
+      } satisfies SpanQueryFilters),
+      fields: [`avg(${SpanFields.SPAN_DURATION})`],
+    },
+    Referrer.SAMPLES_CACHE_TRANSACTION_DURATION
+  );
+
+  const result = useEap ? eapResult : metricsResult;
+  const finalData: Array<{[`avg(span.duration)`]: number}> = useEap
+    ? eapResult.data
+    : metricsResult.data.map(row => ({
+        'avg(span.duration)': row[`avg(${MetricsFields.TRANSACTION_DURATION})`],
+      }));
+
+  return {...result, data: finalData};
```

Comment:
Nit: should we put a memo here? It'll create a new obj ref

Reply:
Hmm, good point, I'll look into that in a separate pr